llvm/lib/Analysis/LoopAccessAnalysis.cpp

```diff
@@ -100,6 +100,8 @@ static cl::opt<unsigned> MemoryCheckMergeThreshold(
 
 /// Maximum SIMD width.
 const unsigned VectorizerParams::MaxVectorWidth = 64;
+/// Maximum LMUL factor.
+const unsigned VectorizerParams::MaxVectorLMUL = 8;
```

Comment:
Is this upper bound RISCV specific?

Reply:
AFAIU yes, stemming from RISCVV having 32 architected vector registers, which become 4 with LMUL = 8. According to https://github.com/riscv/riscv-v-spec/blob/master/v-spec.adoc, LMUL can be 1, 2, 4, 8 and also 1/2, 1/4, 1/8.

Reply:
Currently, yes. I think, we can sink it to TTI later, if some support for other target will be required.